pkg/monitortests/client/api_unreachable.go

```diff
@@ -0,0 +1,164 @@
+		return err
+	}
+	routeClient, err := routeclient.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return err
+	}
+
+	client, err := utilmetrics.NewPrometheusClient(ctx, kubeClient, routeClient)
+	if err != nil {
+		return err
+	}
+
+	kubeSvc, err := kubeClient.CoreV1().Services(metav1.NamespaceDefault).Get(ctx, "kubernetes", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to retrieve cluster IP from kubernetes.default.svc - %v", err)
+	}
+
+	mt.monitor = &monitor{
+		query: &metrics.PrometheusQueryRunner{
+			Client:      client,
+			QueryString: `sum(rate(rest_client_requests_total{code="<error>"}[1m])) by(host)`,
+			Step:        time.Minute,
+		},
+		analyzer: metrics.RateSeriesAnalyzer{},
+		builder: &intervalBuilder{
+			serviceNetworkIP: kubeSvc.Spec.ClusterIP,
+		},
+	}
+
+	framework.Logf("monitor[%s]: monitor initialized, service-network-ip: %s", MonitorName, kubeSvc.Spec.ClusterIP)
+	return nil
+}
+
+func (mt *APIUnReachableFromClientMonitorTest) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	m := mt.monitor
+	if m == nil {
+		return monitorapi.Intervals{}, nil, fmt.Errorf("monitor test is not initialized")
+	}
+
+	if err := m.analyzer.Analyze(ctx, m.query, beginning, end, m.builder); err != nil {
+		return monitorapi.Intervals{}, nil, err
+	}
+	return m.builder.intervals, nil, nil
+}
+
+func (*APIUnReachableFromClientMonitorTest) ConstructComputedIntervals(ctx context.Context, startingIntervals monitorapi.Intervals, recordedResources monitorapi.ResourcesMap, beginning, end time.Time) (monitorapi.Intervals, error) {
+	return nil, nil
+}
+
+func (*APIUnReachableFromClientMonitorTest) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
+	return nil, nil
+}
+
+func (*APIUnReachableFromClientMonitorTest) WriteContentToStorage(ctx context.Context, storageDir, timeSuffix string, finalIntervals monitorapi.Intervals, finalResourceState monitorapi.ResourcesMap) error {
+	return nil
+}
+
+func (*APIUnReachableFromClientMonitorTest) Cleanup(ctx context.Context) error {
+	return nil
+}
+
+type intervalBuilder struct {
+	serviceNetworkIP string
+	locator          monitorapi.Locator
+	intervals        monitorapi.Intervals
+}
+
+func (b *intervalBuilder) Name() string { return MonitorName }
+func (b *intervalBuilder) StartSeries(metric prometheustypes.Metric) {
+	source := b.source(metric)
+	b.locator = monitorapi.NewLocator().WithAPIUnreachableFromClient(source)
+}
+func (b *intervalBuilder) EndSeries() { b.locator = monitorapi.Locator{} }
+
+func (b *intervalBuilder) NewDisruptionInterval(metric prometheustypes.Metric, start, end *prometheustypes.SamplePair) {
+	startTime := start.Timestamp.Time()
+	endTime := end.Timestamp.Time()
+	if start == end {
+		// a disruption window with one sample
+		endTime = end.Timestamp.Time().Add(time.Minute)
```

Comment:
since we scrape every 30s, a `1s` granularity is not very meaningful here, i think. I am approximating the interval to [t-30s ... t+30s] for now. Thoughts?

Reply:
Ah that makes sense, thanks. Fine as is then.